File: src/finance_analysis/interfaces/api/v1/schemas/notifications.py
"""Message center responses. Delivery state is deliberately absent."""

from datetime import datetime
from typing import Optional

from pydantic import BaseModel, ConfigDict, field_validator

from finance_analysis.core.time import coerce_aware_utc


class NotificationPreview(BaseModel):
    id: int
    uid: Optional[int]
    title: str
    content_preview: str
    route_type: str
    severity: str
    created_at: datetime

    _utc = field_validator("created_at")(coerce_aware_utc)


class NotificationListResponse(BaseModel):
    items: list[NotificationPreview]
    total: int
    page: int
    page_size: int


class NotificationDetail(BaseModel):
    model_config = ConfigDict(from_attributes=True)

    id: int
    title: str
    content: str
    route_type: str
    severity: str
    created_at: datetime

    _utc = field_validator("created_at")(coerce_aware_utc)

File: web/src/api/notifications.ts
import apiClient from './index';
import { toCamelCase } from './utils';

export interface NotificationPreview {
  id: number;
  uid: number | null;
  title: string;
  contentPreview: string;
  routeType: string;
  severity: string;
  createdAt: string;
}
export interface NotificationDetail extends Omit<NotificationPreview, 'uid' | 'contentPreview'> {
  content: string;
}
export interface NotificationQuery {
  page?: number;
  page_size?: number;
  keyword?: string;
  route_type?: string;
  severity?: string;
  start_time?: string;
  end_time?: string;
}
export interface NotificationList {
  items: NotificationPreview[];
  total: number;
  page: number;
  pageSize: number;
}
export const notificationsApi = {
  async list(params: NotificationQuery = {}) {
    const { data } = await apiClient.get('/api/v1/notifications', { params });
    return toCamelCase<NotificationList>(data);
  },
  async detail(id: number) {
    const { data } = await apiClient.get(`/api/v1/notifications/${id}`);
    return toCamelCase<NotificationDetail>(data);
  },
};
